<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1743</width>
    <height>990</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1,2">
    <item row="0" column="2">
     <widget class="QGraphicsView" name="graphics_view"/>
    </item>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>15</number>
        </property>
        <property name="topMargin">
         <number>50</number>
        </property>
        <property name="rightMargin">
         <number>15</number>
        </property>
        <item>
         <widget class="QLabel" name="label_12">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Presets:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="leaf_button">
          <property name="text">
           <string>Leaf</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="thistle_button">
          <property name="text">
           <string>Thistle</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="coral_button">
          <property name="text">
           <string>Coral</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="bush_button">
          <property name="text">
           <string>Bush</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="spruce_button">
          <property name="text">
           <string>Spruce</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="binary_tree_button">
          <property name="text">
           <string>Binary Tree</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="foxtail_button">
          <property name="text">
           <string>Foxtail</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="sprig_button">
          <property name="text">
           <string>Sprig</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="algae_button">
          <property name="text">
           <string>Algae</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="birch_button">
          <property name="text">
           <string>Birch</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="rings_button">
          <property name="text">
           <string>Rings</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="cross_button">
          <property name="text">
           <string>Cross</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="koch_snowflake_button">
          <property name="text">
           <string>Koch Snowflake</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="sierpinski_arrowhead_button">
          <property name="text">
           <string>Sierpinski Arrowhead</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="hilbert_curve_button">
          <property name="text">
           <string>Hilbert Curve</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="board_button">
          <property name="text">
           <string>Board</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="crystal_button">
          <property name="text">
           <string>Crystal</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="quadratic_snowflake_button">
          <property name="text">
           <string>Quadratic Snowflake</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="quadratic_koch_island_button">
          <property name="text">
           <string>Quadratic Koch Island</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pentaplexity_button">
          <property name="text">
           <string>Pentaplexity</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="levy_curve_button">
          <property name="text">
           <string>Levy Curve</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="krishna_anklet_button">
          <property name="text">
           <string>Krishna Anklet</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <property name="rowWrapPolicy">
       <enum>QFormLayout::DontWrapRows</enum>
      </property>
      <property name="labelAlignment">
       <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
      </property>
      <property name="formAlignment">
       <set>Qt::AlignHCenter|Qt::AlignTop</set>
      </property>
      <property name="verticalSpacing">
       <number>21</number>
      </property>
      <property name="leftMargin">
       <number>20</number>
      </property>
      <property name="topMargin">
       <number>50</number>
      </property>
      <property name="rightMargin">
       <number>40</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Axiom:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="axiom_textbox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="placeholderText">
         <string>Enter a starting state...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Initial Angle:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="angle_spinbox">
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Iterations:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QSpinBox" name="iterations_spinbox">
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Line Length:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QDoubleSpinBox" name="line_length_spinbox">
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Line Width</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QDoubleSpinBox" name="line_width_spinbox">
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Rules:</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <layout class="QFormLayout" name="rule_form_layout">
        <property name="horizontalSpacing">
         <number>15</number>
        </property>
        <property name="verticalSpacing">
         <number>10</number>
        </property>
       </layout>
      </item>
      <item row="11" column="1">
       <widget class="QPushButton" name="add_rule_button">
        <property name="text">
         <string>Add Rule</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QDoubleSpinBox" name="scale_factor_spinbox">
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QDoubleSpinBox" name="width_increment_spinbox">
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Width Increment</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Scale Factor</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="fill_color_combobox"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Fill Color</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="angle_increment_spinbox">
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>90.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Turn Angle</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="turn_increment_spinbox">
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Turn Increment</string>
        </property>
       </widget>
      </item>
      <item row="15" column="0">
       <widget class="QPushButton" name="center_output_button">
        <property name="text">
         <string>Center Output</string>
        </property>
       </widget>
      </item>
      <item row="15" column="1">
       <widget class="QPushButton" name="generate_button">
        <property name="text">
         <string>Generate!</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1743</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
